platform-sdk/docs/base/test-support/test-support.md

```diff
@@ -61,4 +61,39 @@ executed in isolation (see `org.junit.jupiter.api.parallel.Isolated`).
 ### Context support
 
 The `@WithContext`, `@WithGlobalContext`, `@WithThreadLocalContext` annotation are documented in the chapter of 
-the [Context API](./../context/context.md).
\ No newline at end of file
+the [Context API](./../context/context.md).
+
+### Executor support
+
+The `@WithTestExecutor` annotation is used to inject a `TestExecutor` instance into a test. It can be applied at the
+This utility class is designed to facilitate concurrent testing in multi-threaded environments, making it easier to manage and synchronize tasks in your tests.
+class or method level to enable concurrent testing using a test executor. 
+
```

Comment:
+ It will always wait until all tasks are done. It is executed in isolation. By doing so 2 tests won't use the underlying thread pool in parallel